### PR TITLE
Fixes toBinaryJavaName

### PR DESCRIPTION
**Pull request by errt**
Its semantics are given by https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/Class.html#getName()

### CODE REVIEW

File: OPAL/br/src/main/scala/org/opalj/br/Type.scala [hunk 1]
```diff
@@ -555,7 +548,7 @@ sealed trait CTIntType extends Type {
  */
 case object CTIntType extends CTIntType {
     final val id = Int.MinValue + 1
-    def toBinaryJavaName: String = throw new UnsupportedOperationException()
+    override def toBinaryJavaName: String = throw new UnsupportedOperationException()
```
Comment (maximilianruesch) [inline OPAL/br/src/main/scala/org/opalj/br/Type.scala:551]: Why should int-like types not support this getter? They do, just like every other type (except array)
image

File: OPAL/br/src/main/scala/org/opalj/br/Type.scala [hunk 2]
```diff
@@ -188,10 +188,7 @@ sealed trait Type extends UIDValue with Ordered[Type] {
      * Returns the binary name of this type as used by the Java runtime. Basically
      * returns the same name as produced by `Class.getName`.
```
Comment (maximilianruesch) [inline OPAL/br/src/main/scala/org/opalj/br/Type.scala]:
```suggestion
     * Returns the binary name of this type as used by the Java runtime.
     * Returns the same name as produced by `Class.getName`.
```
Lets update the comment so it says the intended truth.

**Review by maximilianruesch — CHANGES_REQUESTED** [review]
Unsure about a small part.

Can we / should we add tests for this?

File: OPAL/br/src/main/scala/org/opalj/br/Type.scala (hunk 1 above)
Comment (errt) [reply OPAL/br/src/main/scala/org/opalj/br/Type.scala:551]: They do? They inherit the default implementation toBinaryJavaName = toJava from `Type`. The line you marked is only the companion object that represents all computational int types and thus has no individual binary representation.

Comment (maximilianruesch) [reply OPAL/br/src/main/scala/org/opalj/br/Type.scala:551]: Damn didnt see that it was the companion object. Sometimes these confuse me when I do not have IDE support...

This is fine then.

**Review by maximilianruesch — APPROVED** [review]
deemed-excellent

**Pull request merged**